tests/components/integration/test_sensor.py

```diff
@@ -271,42 +234,7 @@ async def test_restore_unavailable_state_failed(
 
     state = hass.states.get("sensor.integration")
     assert state
-    assert state.state == STATE_UNAVAILABLE
-
-
-async def test_restore_state_failed(hass: HomeAssistant) -> None:
-    """Test integration sensor state is restored correctly."""
-    mock_restore_cache(
-        hass,
-        (
-            State(
-                "sensor.integration",
-                "INVALID",
-                {
-                    "last_reset": "2019-10-06T21:00:00.000000",
-                },
-            ),
-        ),
-    )
-
-    config = {
-        "sensor": {
-            "platform": "integration",
-            "name": "integration",
-            "source": "sensor.power",
-        }
-    }
-
-    assert await async_setup_component(hass, "sensor", config)
-    await hass.async_block_till_done()
-
-    state = hass.states.get("sensor.integration")
-    assert state
-    assert state.state == "unknown"
-    assert state.attributes.get("unit_of_measurement") is None
-    assert state.attributes.get("state_class") is SensorStateClass.TOTAL
-
-    assert "device_class" not in state.attributes
+    assert state.state == STATE_UNKNOWN
```

Comment:
Shouldn't it be STATE_UNAVAILABLE ?

Reply:
The test relied on the old restore code to set it to STATE_UNAVAILABLE based on the data in the old state. The new code does not look at the stored state once there is extra data available. Only way it would have triggered this was if the old extra state data was invalid.

Reply:
we should change the test name then